app/javascript/mastodon/utils/debounce.ts

```diff
@@ -0,0 +1,23 @@
+import type { AppDispatch } from 'mastodon/store';
+
+export const debounceWithDispatchAndArguments = <T>(
+  fn: (dispatch: AppDispatch, ...args: T[]) => void,
+  { delay = 100 },
+) => {
+  let argumentBuffer: T[] = [];
+  let timeout: ReturnType<typeof setTimeout>;
+  let dispatchBuffer: AppDispatch;
+
+  const flush = () => {
+    const tmpBuffer = argumentBuffer;
+    argumentBuffer = [];
+    fn(dispatchBuffer, ...tmpBuffer);
+  };
+
+  return (dispatch: AppDispatch, ...args: T[]) => {
+    dispatchBuffer = dispatch;
+    argumentBuffer.push(...args);
+    clearTimeout(timeout);
+    timeout = setTimeout(flush, delay);
+  };
```

Comment:
I think this behaves differently than one would expect of `debounce`, as if multiple requests come up within `delay` of each other, the underlying function won't actually be called after `delay` milliseconds after the last call, whereas e.g. loadash's `debounce` would mean the underlying function is called multiple times but no more than once every `delay` milliseconds.

Reply:
I think you may be thinking of `throttle`? Although lodash `debounce` has options to e.g. execute immediately on first call, etc. But I figured this is a very specialized version and it makes sense for this use case.

Reply:
I think what might make sense is to add a "maximum arguments" counter and execute whenever that maximum is reached, since the relationships API has a limit for number of IDs you can pass it.